src/evolution.jl

```diff
@@ -20,37 +21,100 @@ include("constants.jl")
     cutoff = truncation threshold for the SVD in MPS representation (unitless and dimensionless)
     periodic = boolean indicating whether boundary conditions should be periodic
+    Expected (CGS) units of the quantities defined in the files in tests directory that are being used in the evolve function.                                                                   
+    s = site index array (dimensionless and unitless)          
+    N = array of no.of neutrinos contained on each site (dimensionless and unitless)
+    B = array of normalized vector related to mixing angle in vacuum oscillations (dimensionless constant)
+    N_sites = Total no.of sites (dimensionless and unitless)
+    Δx = length of the box of interacting neutrinos at a site (cm)
+    Δm² = difference in mass squared (erg^2)
+    p = array of momentum vectors (erg)
+    x = array of positions of sites (cm)
+    Δp = width of shape function (cm)
+    shape_name = name of the shape function (string) ["none","triangular","flat_top"]
+    τ = time step (sec)
+    energy_sign = array of sign of the energy (1 or -1): 1 for neutrinos and -1 for anti-neutrinos
+    maxdim = max bond dimension in MPS truncation (unitless and dimensionless)
+    cutoff = truncation threshold for the SVD in MPS representation (unitless and dimensionless)
+    periodic = boolean indicating whether boundary conditions should be periodic
 """
 
 # This file generates the evolve function which evolves the ψ state in time and computes the expectation values of Sz at each time step, along 
 # with their survival probabilities. The time evolution utilizes the unitary operators created as gates from the create_gates function.
 # The <Sz> and Survival probabilities output from this function are unitless. 
-function evolve(s, τ, N, B, L, N_sites, Δx, Δm², p, x, Δp, theta_nu, ψ, shape_name, energy_sign, cutoff, maxdim, datadir, t1, t2, ttotal, save_data::Bool, periodic=true)
-
-    # Create empty array s to...
-    Sz_array = Float64[] # to store sz values 
-    Sy_array = Float64[] # to store sy values 
-    Sx_array = Float64[] # to store sx values 
-    t_array = [] # to store t values 
-    prob_surv_array = Float64[]   # to store survival probability values 
+function evolve(s, τ, N, B, L, N_sites, Δx, Δm², p, x, Δp, theta_nu, ψ, shape_name, energy_sign, cutoff, maxdim, datadir, t1, t2, ttotal, chkptdir, checkpoint_every, do_recover, recover_type, recover_iteration, save_data::Bool, periodic=true)
+
+    t_initial = 0.0
+    iteration = 0
+    t_recover = 0.0  # Variable to store the initial recovery time 
+
+    if do_recover
+        
+        if recover_type == "auto"
+            println("auto recovery: recovering from last iteration")
+            recover_iteration = -1
+            checkpoint_files = readdir(chkptdir)
+            checkpoint_files = filter(f -> endswith(f, ".h5"), checkpoint_files)
+            checkpoint_files = sort(checkpoint_files)
```

Comment:
Should the checkpoint file string just be an optional argument instead of searching for them? This would give you the additional flexibility to restart from any file, and not tie you to only restarting from the latest checkpoint in your current directory.

Reply:
Perhaps that's a comment for the "manual" section below. You can just pass the filename instead of the iteration.

Reply:
Also, both functions share the actual "recover_checkpoint_hdf5" function, so that (and later) lines can be removed from the recover_type if statements.